xpra/net/ssh.py

```diff
@@ -930,17 +932,18 @@ def ssh_exec_connect_to(display_desc, opts=None, debug_cb=None, ssh_fail_cb=ssh_
                 #sshpass -e ssh ...
                 cmd.insert(0, sshpass_command)
                 cmd.insert(1, "-e")
-                if env is None:
-                    env = os.environ.copy()
                 env["SSHPASS"] = password
                 #the password will be used by ssh via sshpass,
                 #don't try to authenticate again over the ssh-proxy connection,
                 #which would trigger warnings if the server does not require
                 #authentication over unix-domain-sockets:
                 opts.password = None
                 del display_desc["password"]
-        if env:
-            kwargs["env"] = env
+        # Scrub out env vars our wrapper set so we don't potentially upset the ssh process
+        for v in ["PYTHONHOME", "PYTHONPATH", "PYTHON", "DYLD_LIBRARY_PATH", "LD_PRELOAD", "LD_LIBRARY_PATH"]:
```

Comment:
Does the python wrapper really set all of these?
I can understand scrubbing `PYTHON*` but the `*LD*` ones could be left alone?
(if the user has set them, he may have a reason)

Reply:
It sets DYLD_LIBRARY_PATH on OSX, so I figured it may set LD_* on FreeBSD/Linux/etc, however I haven't checked.
It would be nice if there was a way to get the "at launch" environment but I guess that's not feasible.

This does only affect the environment for the launched SSH so I thought it would be a reasonable compromise but I agree it's not ideal.

Reply:
Ah, so you're on OSX.
The client goes through this script:
https://github.com/Xpra-org/xpra/blob/master/packaging/MacOS/Helpers/PythonExecWrapper

So I see 2 possible solutions:
* modify the script to insert xpra's paths, but preserve any values the variables may already have - this may or may not be enough to make things work as unset variables will still be set to the new value..
* merge your change but only sanitize the environment on MacOS (`if OSX: ..`)

---

FYI: On Fedora, none of these variables are set by the python interpreter:
```
$ python3 -c 'import os;print("\n".join("%s=%s" % (k, v) for k,v in os.environ.items() if k.find("PY")>=0 or k.find("LD")>=0))'
```
(empty output)

I haven't tested MS Windows yet.

Reply:
Ahh, I thought it was a generic tool not something specific to Xpra.
I will try the first approach and see how I go.